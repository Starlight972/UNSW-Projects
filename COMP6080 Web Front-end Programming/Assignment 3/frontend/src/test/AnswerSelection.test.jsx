import React from 'react';
import { shallow } from 'enzyme';

// MUI components
import { Button } from '@mui/material';

// own component
import AnswerSelection from '../components/AnswerSelection';

describe('<AnswerSelection />', () => {
  let wrapper;
  const answer = { answerContext: 'Answer_1', correct: false };

  it('renders without crashing', () => {
    wrapper = shallow(<AnswerSelection answer={answer} />);
  })

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have text', () => {
    expect(wrapper.find('.answerText').text()).toBe('Answer_1');
  })

  it('should render one component', () => {
    expect(wrapper.find(Button)).toHaveLength(1);
  })
})
