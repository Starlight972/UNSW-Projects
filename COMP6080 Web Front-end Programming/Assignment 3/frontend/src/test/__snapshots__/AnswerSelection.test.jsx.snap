// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnswerSelection /> matches snapshot 1`] = `ShallowWrapper {}`;
